src/engine/audio/SoundCodec.cpp

```diff
@@ -68,32 +93,24 @@ AudioData LoadSoundCodec(std::string filename)
 		}
 	}
 
-	// if filename does not have extension or there is no file with such extension
-	// or if there is no codec available for this file format,
-	// try and find a suitable match using all the sound file formats supported
-	// prioritize with the pak priority
-	int bestLoader = -1;
-	const FS::PakInfo* bestPak = nullptr;
-	std::string strippedname = FS::Path::StripExtension(filename);
+	// if the file isn't there, maybe the file path did not have any extension,
+	// and it may be possible the file has a dot in its name that was mistakenly
+	// taken as an extension
+	int bestLoader = FindSoundLoader(filename);
 
-	for (int i = 0; i < numSoundLoaders; i++)
+	if (!ext.empty() && bestLoader == -1)
 	{
-		std::string altName = Str::Format("%s%s", strippedname, soundLoaders[i].ext);
-		const FS::PakInfo* pak = FS::PakPath::LocateFile(altName);
+		// if the file isn't there, maybe its filename does not have extension
```

Comment:
This comment seems out of place since this block can only work if the filename does have an extension.